src/operations/execute_operation.ts

```diff
@@ -89,19 +89,13 @@ export function executeOperation<
     // that are not explicitly provided with a session.
     let session: ClientSession | undefined = operation.session;
     let owner: symbol | undefined;
-    if (topology.hasSessionSupport()) {
-      if (session == null) {
-        owner = Symbol();
-        session = topology.startSession({ owner, explicit: false });
-      } else if (session.hasEnded) {
-        return cb(new MongoExpiredSessionError('Use of expired sessions is not permitted'));
-      } else if (session.snapshotEnabled && !topology.capabilities.supportsSnapshotReads) {
-        return cb(new MongoCompatibilityError('Snapshot reads require MongoDB 5.0 or later'));
-      }
-    } else if (session) {
-      // If the user passed an explicit session and we are still, after server selection,
-      // trying to run against a topology that doesn't support sessions we error out.
-      return cb(new MongoCompatibilityError('Current topology does not support sessions'));
+    if (session == null) {
+      owner = Symbol();
+      session = topology.startSession({ owner, explicit: false });
+    } else if (session.hasEnded) {
```

Comment:
With these changes we assume `logicalSessionTimeoutMinutes` is always there and a number, and if it is not that's a server bug. With respect to the default of 10 minutes - I'm not sure where that came from and it's not in the spec. The spec only states a 30 minute timeout on the server side which is configurable on the server for when the session is idle - so if the session is not idle I'd imagine for long running operations like ETL jobs they could be valid for very long durations (especially with explicit sessions). I looked in the Java driver and they default it to 30, so I'm not sure why we default to 10. Maybe a question for next week when Jeff is back.

Reply:
That's not an assumption that is safe to make I think, the rule of determining `logicalSessionTimeoutMinutes` is that there must be data bearing servers, and you pick the minimum between all nodes, unless _any_ node has a null value than you set it to null for the whole topology. 

Using our mlaunch helper script I can get an undefined logicalSessionTimeoutMinutes like so:
```
> var client = await mdb.MongoClient.connect('mongodb://bob:pwd123@localhost:31003?directConnection=true')
> client.topology.logicalSessionTimeoutMinutes
undefined
```
The node at 31003 is an arbiter

Reply:
My thought here was that if there weren't any data bearing servers then server selection would fail when executing the command and there would be no need for the extra server selection to happen ahead of time, since its purpose was only to determine session support. We would not select an arbiter, unknown, rsghost , or rsother to run a command against - the only thing getting sent there is the `hello` from the monitor which does not send a session. If any data bearing server in the cluster had a null `logicalSessionTimeoutMinutes` that would indicate to me that it was a mixed version cluster and that node was pre 3.6 which would error now with the wire version bump. But it seems there a lot of potential unknowns here after this was all brought up. So I'm thinking I just roll back this PR to the single spot to fix this issue - the check in the cursor. (And just leave the remaining checks in for the time being). Thoughts?

Reply:
Yea that sounds good to me, and actually another thing we could try is instead of removing the check what about reusing your "reselect same server" logic from the getMore fix awhile back inside next before we call hasSessionSupport. The problem I think originates from the fact that `hasSessionSupport` is a value that changes depending on the current state of the topology, running a server selection will make sure that value is updated before the code proceeds, the other option is that hasSessionSupport is something that we might want to refactor to be async, but I think the selection change would be easier for now. Want to give that a try?

Reply:
When direct connecting to an arbiter on this PR vs main, there is an LSID being attached even when logicalSessionTimeoutMinutes is null, "server selection would fail when executing the command" isn't going to be true for all commands, the point of connecting to an arbiter is to do management commands on it, So I think we need to only change the cursor logic to not impact all commands / topology scenarios
```
NODE-3521 {
  ping: 1,
  lsid: {
    id: new Binary(Buffer.from("80990b6ec8d4451082d3c3361e901479", "hex"), 4)
  },
  '$db': 'test'
}
```

```
main branch { ping: 1, '$db': 'test' }
```

Reply:
I liked the idea of doing server selection in this case so I reverted all the changes and introduced the `shouldCheckForSessionSupport` logic into the cursor's `next` method. Selecting the same server wouldn't be valid in this edge case as no server has actually been selected yet (the initial find command hasn't been run) but I think before we check for session support we always need to run server selection if the topology is in a state that it cannot be determined. I also think this brings us in line with the spec - we always should do this when checking.

Reply:
I've also just verified that this latest change also addresses the issue on lambda.